src/main/java/org/sagebionetworks/bridge/spring/controllers/HealthDataEx3Controller.java

```diff
@@ -0,0 +1,151 @@
+    public StatusMessage deleteRecordsForUser(@PathVariable String userId) {
+        UserSession session = getAuthenticatedSession(SUPERADMIN);
+
+        String healthCode = accountService.getHealthCodeForAccount(AccountId.forId(session.getAppId(), userId));
+        if (healthCode == null) {
+            throw new EntityNotFoundException(StudyParticipant.class);
+        }
+
+        healthDataEx3Service.deleteRecordsForHealthCode(healthCode);
+
+        return new StatusMessage("Health data has been deleted for participant");
+    }
+
+    /** Retrieves the record for the given ID. */
+    @GetMapping(path="/v1/admin/exporter3/healthdata/{recordId}")
+    public HealthDataRecordEx3 getRecord(@PathVariable String recordId) {
+        getAuthenticatedSession(SUPERADMIN);
+        HealthDataRecordEx3 record = healthDataEx3Service.getRecord(recordId).orElseThrow(() ->
+                new EntityNotFoundException(HealthDataRecordEx3.class));
+
+        // Write record ID into the metrics, for logging and diagnostics.
+        Metrics metrics = getMetrics();
+        if (metrics != null) {
+            metrics.setRecordId(record.getId());
+        }
+
+        return record;
+    }
+
+    /** Retrieves all records for the given user and time range. */
+    @GetMapping(path="/v1/admin/exporter3/participants/{userId}/healthdata")
+    public ResourceList<HealthDataRecordEx3> getRecordsForUser(@PathVariable String userId,
+            @RequestParam(required = false) String createdOnStart,
+            @RequestParam(required = false) String createdOnEnd) {
+        UserSession session = getAuthenticatedSession(SUPERADMIN);
+
+        String healthCode = accountService.getHealthCodeForAccount(AccountId.forId(session.getAppId(), userId));
+        if (healthCode == null) {
+            throw new EntityNotFoundException(StudyParticipant.class);
+        }
+
+        DateTime createdOnStartDateTime = BridgeUtils.getDateTimeOrDefault(createdOnStart, null);
+        DateTime createdOnEndDateTime = BridgeUtils.getDateTimeOrDefault(createdOnEnd, null);
+        List<HealthDataRecordEx3> recordList = healthDataEx3Service.getRecordsForHealthCode(healthCode,
+                createdOnStartDateTime, createdOnEndDateTime);
+
+        return new DateTimeRangeResourceList<>(recordList)
+                .withRequestParam("userId", userId)
+                .withRequestParam(ResourceList.START_TIME, createdOnStart)
+                .withRequestParam(ResourceList.END_TIME, createdOnEnd);
+    }
+
+    /** Retrieves all records in the current app for the given time range. */
+    @GetMapping(path="/v1/admin/exporter3/apps/self/healthdata")
+    public ResourceList<HealthDataRecordEx3> getRecordsForCurrentApp(
+            @RequestParam(required = false) String createdOnStart,
+            @RequestParam(required = false) String createdOnEnd) {
+        UserSession session = getAuthenticatedSession(SUPERADMIN);
```

Comment:
Not worker accounts? The endpoint URL also encodes who can access the endpoint in /admin/ rather than just naming the resource, so you would have a harder time changing this later. Are there any other kinds of records than health data records? Is this a set of  /v1/healthdata/ endpoints?

Reply:
These APIs are only exposed for integration tests, so it should be pretty easy to change these or remove these later on.